[KS] ProgressMeter Bugfix transparancy and responsibility

qdittrich opened the pull request:
This bugfix for the ProgressMeter KitchenSink component is step one of the bugfix- / feature-bundle, mentioned in the ticket below.
This bugfix has only effect on the "mini" type of the component. The  wording "mini" does not longer refer to the size, but is meant as an shortcut for "minimalistic" as told in the modified description.

https://mantis.ilias.de/view.php?id=23654

qdittrich committed:
Merge remote-tracking branch 'origin/trunk' into fix-23654

qdittrich committed:
WIP Bugfix 23654 step 1

qdittrich committed:
WIP Bugfix 23654 step 2

qdittrich committed:
Merge remote-tracking branch 'origin/trunk' into fix-23654

qdittrich committed:
WIP Bugfix 23654 Updated UnitTests

Amstutz commented:
@qdittrich thx for this fix. However, one test is failing, please fix:
![image](https://user-images.githubusercontent.com/1866896/47671499-e86cbc00-dbaf-11e8-8709-d8cdbd9209dd.png)

qdittrich committed:
WIP Bugfix 23654 Updated UnitTests

qdittrich commented:
I'm currently not able to properly fix the PHPUnit test, because since any update some time ago, I cannot get the PHPUnit tests running on my machine anymore. Even reinstalling PHPUnit and composer does not work.

I only changed one file (I added some spaces at a string) and now Travis throws lots of errors on other files, I never touched. The last time there was an similiar behavior about that "Dropzone" components; do you remember @Amstutz ?
However, all "ProgressMeter" tests pass. The last failure was because of missing spaces inside to parts of the test string. I added these spaces and now this test seems to pass.

Amstutz commented on src/UI/templates/default/Chart/chart.less:
```diff
@@ -524,4 +466,55 @@
+  .il-chart-progressmeter-mini-container {
+    position: relative;
+    min-width: 20px;
+    max-width: 250px;
```
Where do those values come from? How are they calculated/determined?

Amstutz commented on src/UI/templates/default/Chart/chart.less:
```diff
@@ -524,4 +466,55 @@
+      display: block;
+      max-width: 80%;
+      min-width: 20px;
+      max-height: 250px;
```
Same values as above, introduce variables.

Amstutz commented on src/UI/templates/default/Chart/chart.less:
```diff
@@ -524,4 +466,55 @@
+      .il-chart-progressmeter-circle-bg {
+        stroke: @il-chart-progressmeter-background;
+        fill: none;
+        stroke-width: 5;
```
Definitely introduce a variable for this one (e.g. il-chart-progressmeter-mini-stroke-width)

Amstutz commented on src/UI/templates/default/Chart/chart.less:
```diff
@@ -524,4 +466,55 @@
+      }
+      .il-chart-progressmeter-circle {
+        fill: none;
+        stroke-width: 5;
```
Again the variable

Amstutz commented on src/UI/templates/default/Chart/chart.less:
```diff
@@ -524,4 +466,55 @@
+      .il-chart-progressmeter-needle-mini {
+        stroke: @il-chart-progressmeter-mini-marker-color;
+        fill: none;
+        stroke-width: 4px;
```
Make this dependend from the above variable or maybe event introduce a new one.

Amstutz commented:
Hi @qdittrich Thx for fixing the failing tests. The reason travis was failing in other tests was kind of "bad luck" from your side. You committed the fix of your tests while other tests where broken due to work of another dev (see build history in travis). This tests have been fixed by the according dev. in the meanwhile. All I needed to do was to hit rerun for your PR in travis (I am not sure if you can do that):
![image](https://user-images.githubusercontent.com/1866896/47698205-aa52b500-dc0e-11e8-80f4-4d038299f793.png)

and all the lights for your test did turn green. Since this is a regular bugfix, there would be no issue with merging this now. However there are some todos. Please fix them before we can proceed:

- [ ] You changed the description of your component. This is good, thx. However, such changes (especially fundemantal ones like yours) need to be approved by the JF). Take this out for the moment and open a new PR with just the change of the description so we can merge the technical fix know and discuss your proposal of the semantic change later in the JF with a new PR (propably together with the other changes you are intending to perform for the other progress meter.
- [ ] Currenlty the new mini version does not look right in the examples (tested on safari and chrome). In my case they look like this:
![image](https://user-images.githubusercontent.com/1866896/47699108-1f73b980-dc12-11e8-8819-71e0f1c324f6.png)

- [ ] Some minor points with the less (see my comments).

[Edit:] I just found the reason, the less was not recompiled. So the second point is easily fixed by just recompiling less to include your changes in the css and commit.

qdittrich commented:
First of all: Thank you for your comments and your help! 👍 

I will revert the component description and set it in the "feature" PR again, as you mentioned above. I did not know that changes in this place need a JF decision.

About the styling: I will introduce some new variables.
For explanation about the sizes:
The min-width and max-width / max-height are set because it seemed to look a bit distorted if it gets smaller then 20px or bigger then 250px. Also, if it gets smaller then 20px, you can not clearly see your progress anymore because it is to small (at least for my eyes).

qdittrich committed:
Bugfix 23654 revert component description

qdittrich committed:
Bugfix 23654 introduced new variables

qdittrich committed:
Merge remote-tracking branch 'origin/trunk' into fix-23654

qdittrich commented:
Hi @Amstutz 
I updated the less files, introduced some variables and recompiled the delos less. Also I reverted the component description and merged the trunk, so the PHPUnit tests should pass now.

Amstutz commented:
Hi @qdittrich almost ready to merge ;-). There is a new issue, double check the CSS file, I believe you committed a lot more changes than should originate from you less. This might be due to your configuration of your lessc compiler. Nervertheless, you should only commit css changes that do origin from you own less.

qdittrich committed:
Bugfix 23654 fixed and recompiled delos css

Amstutz commented:
Thx a lot for this fix!
